pkg/compute/store/boltdb/store.go

```diff
@@ -413,6 +421,11 @@ func (s *Store) updateExecutionState(tx *bolt.Tx, request store.UpdateExecutionS
 	localExecutionState.Version += 1
 	localExecutionState.UpdateTime = time.Now()
 
+	// Each time we move into the running state, we want to use up one of the restart attempts
+	if request.NewState == store.ExecutionStateRunning {
+		localExecutionState.Execution.Job.Task().RestartPolicy.Attempts = retryAttempts - 1
+	}
+
```

Comment:
it sounds to me we are doing too much logic inside the data store. A data store should be as simple and stupid as possible, and let the business logic live somewhere else. It should just compare conditions, and validate the integrity of the end state, but not change the models on behalf of the caller. Or else you would need to do this in all data store implementations